CoseSignTool/CoseSign1/Extensions/CoseSign1MessageDetachedSignatureExtensions.cs

```diff
@@ -0,0 +1,134 @@
+    /// </summary>
+    /// <param name="this">The COSE Sign1 Message to evaluate.</param>
+    /// <returns>True if the COSE Sign1 Message is a encoded detached signature; False otherwise.</returns>
+    public static bool IsDetachedSignature(this CoseSign1Message @this) => @this.TryGetDetachedSignatureAlgorithm(out _);
+
+    /// <summary>
+    /// Computes if the encoded detached signature within the COSE Sign1 Message object matches the given artifact stream.
+    /// </summary>
+    /// <param name="this">The COSE Sign1 Message to evaluate.</param>
+    /// <param name="artifactStream">The artifact stream to evaluate.</param>
+    /// <returns>True if the detached signature in the COSE Sign1 Message matches the signature of the artifact stream; False otherwise.</returns>
+    public static bool SignatureMatches(this CoseSign1Message @this, Stream artifactStream)
+    {
+        if (!@this.TryGetHashAlgorithm(out HashAlgorithm? hasher))
+        {
+            Trace.TraceError($"{nameof(SignatureMatches)} failed to extract a valid HashAlgorithm from the provided CoseSign1Message[{@this?.GetHashCode()}]");
+            return false;
+        }
+        ReadOnlyMemory<byte> artifactHash = hasher.ComputeHash(artifactStream);
+        bool equals = @this.Content.Value.Span.SequenceEqual(artifactHash.Span);
+        Debug.WriteLine($"{nameof(SignatureMatches)} compared the two hashes with lengths ({artifactHash.Length},{@this.Content.Value.Length}) for equality and returned {equals}");
+        return equals;
+    }
+
+    /// <summary>
+    /// Computes if the encoded detached signature within the COSE Sign1 Message object matches the given artifact bytes.
+    /// </summary>
+    /// <param name="this">The COSE Sign1 Message to evaluate.</param>
+    /// <param name="artifactBytes">The artifact bytes to evaluate.</param>
+    /// <returns>True if the detached signature in the COSE Sign1 Message matches the signature of the artifact bytes; False otherwise.</returns>
+    public static bool SignatureMatches(this CoseSign1Message @this, ReadOnlyMemory<byte> artifactBytes)
+    {
+        if(!@this.TryGetHashAlgorithm(out HashAlgorithm? hasher))
+        {
+            Trace.TraceError($"{nameof(SignatureMatches)} failed to extract a valid HashAlgorithm from the provided CoseSign1Message[{@this?.GetHashCode()}]");
+            return false;
+        }
+
+        ReadOnlyMemory<byte> artifactHash = hasher.ComputeHash(artifactBytes.ToArray());
+        bool equals = @this.Content.Value.Span.SequenceEqual(artifactHash.Span);
+        Debug.WriteLine($"{nameof(SignatureMatches)} compared the two hashes with lengths ({artifactHash.Length},{@this.Content.Value.Length}) for equality and returned {equals}");
+        return equals;
+    }
+
+    /// <summary>
+    /// Extracts a HashAlgoritm used to compute hashes from a given COSE Sign1 Message object if possible.

```

Comment:
For standard practice, I think "COSE Sign1 Message" is supposed to be one word. If you don't feel like messing with it I'll probably catch it my next PR.

Reply:
Updated in the next push